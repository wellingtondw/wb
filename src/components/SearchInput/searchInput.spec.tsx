import { renderWithTheme } from "../../utils/tests/helper";

import { SearchInput } from ".";
import { fireEvent, screen } from "@testing-library/react";

describe("<SearchInput />", () => {
  it("should be able to render correctly", () => {
    const { container } = renderWithTheme(
      <SearchInput label="label" icon="location" />
    );

    expect(screen.getByText("label")).toBeInTheDocument();
    expect(screen.getByText("label").previousSibling).toHaveClass(
      "icon-location"
    );
    expect(screen.getByRole("button").firstChild).toHaveClass("icon-close");

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should be able to change and clear input value", () => {
    const onInputChange = jest.fn();
    renderWithTheme(
      <SearchInput
        label="label"
        icon="location"
        disabled
        onInputChange={onInputChange}
      />
    );

    const inputElement = screen.getByRole("textbox");
    const buttonClearElement = screen.getByRole("button");

    fireEvent.change(inputElement, { target: { value: "input changed" } });

    expect(inputElement).toHaveValue("input changed");
    expect(onInputChange).toHaveBeenCalledWith("input changed");

    fireEvent.click(buttonClearElement);

    expect(inputElement).toHaveValue("");
  });
});
