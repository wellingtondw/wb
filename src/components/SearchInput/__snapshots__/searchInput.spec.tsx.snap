// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should be able to render correctly 1`] = `
<div
  class="sc-bczRLJ dCOqkO"
>
  <div
    class="sc-hKMtZM enDLmK"
  >
    <span
      class="icon-location"
      style="font-size: 26px;"
    />
    <p>
      label
    </p>
  </div>
  <input
    class="sc-gsnTZi bHaydG"
    type="text"
    value=""
  />
  <button
    class="sc-eCYdqJ bwkndC"
  >
    <span
      class="icon-close"
      style="font-size: 16px;"
    />
  </button>
</div>
`;
